modules/snmp/validators.go

```diff
@@ -0,0 +1,144 @@
+package snmp
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/netdata/go.d.plugin/agent/module"
+)
+
+func appendError(err error, msg string) error {
+	var e error
+	if err == nil {
+		if msg == "" {
+			return nil
+		}
+		e = errors.New(msg)
+	} else {
+		if msg == "" {
+			return err
+		}
+		e = fmt.Errorf("%w %s", err, msg)
+	}
+	return e
+}
+
+func (d Dimension) validateConfig(index_chart, index int) error {
+	var err error
+	err = nil
+	if d.Name == "" {
+		err = appendError(err, fmt.Sprintf("invalid or missing value: charts[%d].dimension[%d].name;", index_chart, index))
+	}
+	if d.OID == "" {
+		err = appendError(err, fmt.Sprintf("missing value: charts[%d].dimension[%d].oid;", index_chart, index))
+	}
+	if d.Algorithm == "" ||
+		(d.Algorithm != string(module.Incremental) &&
+			d.Algorithm != string(module.PercentOfIncremental) &&
+			d.Algorithm != string(module.PercentOfAbsolute)) {
+		err = appendError(err, fmt.Sprintf("invalid or missing value: charts[%d].dimension[%d].algorithm;", index_chart, index))
+	}
+	if d.Multiplier == 0 {
+		err = appendError(err, fmt.Sprintf("integer set to 0: charts[%d].dimension[%d].multiplier;", index_chart, index))
+	}
```

Comment:
An empty/unset value is ok here (Netdata will use an internal default), it is better to have as less mandatory parameters as possible. 0 means no multiplier.

Reply:
I think this check should still be valid. We can initialize multiplier=1 for unset value at init(). But if user exclusively sets it to 0, then it should return error.

Reply:
0 is the zero value for `int`. Does it mean this check will always fail if not explicitly set?

Reply:
I mean, if we need to set default, it should happen before validating. Validator is a way to verify even the default values set by us. So I can check if we can initialize default at some other places like init() function or introduce default struct tags.